### PR TITLE
Improves self hosted login errors

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes: #4945 

This PR tries to improved the code around displaying error messages on login, and improve the login messages show to the user.

To test:

Try different scenarios on login:
 * For a self hosted site:
   * Login to a 404 site
   * Login to a 301 site
   * Login to a WordPress site with a version lower than 4.0
   * Login to sites with basic auth
     * Cancel the basic auth
     * put a wrong credential
     * put the right credential
  * Access sites with https setup correctly
  * Access sites with https setup with self signed certificate.
  
Needs review: @astralbodies and @bummytime

### CODE REVIEW

**Comment by astralbodies** [comment]
Are we considering this a bug fix then to get it into 6.1 or should this be targeted at 6.2/develop? cc: @sendhil

**Comment by SergioEstevao** [comment]
If 6.1 still has a lot of time of internal testing I think this is good to
go in it.

File: WordPress/Classes/Services/BlogService.h [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+extern NSString *const WordPressMinimumVersion;
```
Comment (astralbodies) [inline WordPress/Classes/Services/BlogService.h]: :heart:

File: WordPress/Classes/Services/WordPressXMLRPCAPIFacade.m [hunk 2]
```diff
@@ -23,18 +23,15 @@ - (void)guessXMLRPCURLForSite:(NSString *)url
...
-    } else if ([error.domain isEqual:AFURLRequestSerializationErrorDomain] || [error.domain isEqual:AFURLResponseSerializationErrorDomain]) {
-        NSString *str = [NSString stringWithFormat:NSLocalizedString(@"There was a server error communicating with your site:\n%@\nTap 'Need Help?' to view the FAQ.", nil), [error localizedDescription]];
-        NSDictionary *userInfo = @{NSLocalizedDescriptionKey: str};
-        NSError *err = [NSError errorWithDomain:@"org.wordpress.iphone" code:NSURLErrorBadServerResponse userInfo:userInfo];
-        return err;
     } else {
-        NSDictionary *userInfo = @{NSLocalizedDescriptionKey: NSLocalizedString(@"Unable to find a WordPress site at that URL. Tap 'Need Help?' to view the FAQ.", nil)};
-        NSError *err = [NSError errorWithDomain:@"org.wordpress.iphone" code:NSURLErrorBadURL userInfo:userInfo];
+        NSDictionary *userInfo = @{
+                                   NSLocalizedDescriptionKey: NSLocalizedString(@"Unable to read the WordPress site on that URL. Tap 'Need Help?' to view the FAQ.", nil),
```
Comment (astralbodies) [inline WordPress/Classes/Services/WordPressXMLRPCAPIFacade.m]: > Unable to read the WordPress site on that URL. Tap 'Need Help?' to view the FAQ.
Sounds better if you use "at that URL".

**Comment by astralbodies** [comment]
Several unit tests are failing and need to be updated:

**Comment by sendhil** [comment]
@SergioEstevao  - There are significant changes in here that make me nervous from passing it off as a 'bug fix' to be put into the release branch. I'd re-submit it against develop and we'll properly test it for the next release.

File: WordPress/Classes/Services/WordPressXMLRPCAPIFacade.m (hunk 2 above)
Comment (SergioEstevao) [inline WordPress/Classes/Services/WordPressXMLRPCAPIFacade.m]: I wanted to match the text that shows in the FAQ, probably we will need to update the FAQ itself.

**Comment by SergioEstevao** [comment]
Closing this PR and resubmit a new one for the develop branch.

**Pull request closed without merging**